source/distributed/solution_transfer.cc

```diff
@@ -247,27 +341,90 @@ namespace parallel
       const std::vector<::dealii::Vector<typename VectorType::value_type>>
         dofvalues = Utilities::unpack<
           std::vector<::dealii::Vector<typename VectorType::value_type>>>(
-          data_range.begin(), data_range.end(), /*allow_compression=*/false);
+          data_range.begin(), data_range.end(), has_hp_dof_handler);
```

Comment:
can you re-add the `/* allow_compression= */` comment? I think it is useful to explain why `has_hp...` appears in this argument slot